### ISSUE TITLE
e parsing should read exactly one digit

### ISSUE THREAD

**Issue by Acumane** [moment/moment#6116]
**Describe the bug**
`.format('eHHmm')` returns an expected time string that moment validates as *in*correct.

**To Reproduce**
1. `Moment.tz('America/New_York').format('eHHmm')` produces, say, `21530`
2. But `Moment('21530', 'eHHmm').isValid()` returns `false`

**Expected behavior**
These two results should be (at least intuitively) consistent

**Desktop (please complete the following information):**
 - OS: Windows 11 WSL2
 - Browser: Chromium
 - Version: 112.0.5615.49 (Official Build) (64-bit) 

**Moment-specific environment**
- Vite 4.2.1 w/ Vue 3.2.47 running Node v18.12.1
- Currently running on EDT

Please run the following code in your environment and include the output:
```
Tue Apr 11 2023 02:55:49 GMT-0400 (Eastern Daylight Time)
4/11/2023, 2:55:49 AM
240
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36
2.29.4
```
**Additional context**
`Moment('2-1530', 'e-HHmm')` with any separator is, however, valid.

**Comment by ichernev** [comment]
Yeah, `e` reads any number of digits instead of exactly one. You can submit a PR.

**Comment by shyok21** [comment]
Hey, I’ve opened a PR that addresses this issue: #6318.

The fix checks for compact inputs like `eHHmm` (e.g. `21530`) and expands them internally to `e HHmm` before parsing. This allows the existing parser to work without changes to token logic.

Let me know if you'd prefer a different direction — happy to tweak based on suggestions.
